cmake/tests/cxx17_inline_variable.cpp

```diff
@@ -0,0 +1,12 @@
+//  Copyright (c) 2020 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+inline constexpr int v = 42;
+
+int main()
+{
+    return 0;
+}
```

Comment:
By the way, @sithhell has the same functionality in #4305 (https://github.com/STEllAR-GROUP/hpx/pull/4305/files#diff-b6fbaa7a023d645956d24534dff6909e). The other one looks a bit more comprehensive, although I guess this one would do the job as well.

Reply:
Yes, I saw that. I was planning to consolidate this once one of the PRs is merged.